core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/workflow/add/ValidateAddRequestActivity.java

```diff
@@ -201,15 +207,16 @@ protected Sku findMatchingSku(Product product, Map<String, String> attributeValu
         return matchingSku;
     }
 
-    protected void addSkuToCart(Sku sku, OrderItemRequestDTO orderItemRequestDTO, Product product, CartOperationRequest request) {
+    protected void addSkuToCart(Sku sku, OrderItemRequestDTO orderItemRequestDTO, Product product, CartOperationRequest request) throws InventoryUnavailableException{
         // If we couldn't find a sku, then we're unable to add to cart.
         if (!hasSkuOrIsNonDiscreteOI(sku, orderItemRequestDTO)) {
             handleIfNoSku(orderItemRequestDTO, product);
         } else if (sku == null) {
             handleIfNonDiscreteOI(orderItemRequestDTO);
         } else if (!sku.isActive()) {
-            throw new IllegalArgumentException("The requested skuId (" + sku.getId() + ") is no longer active");
-        } else {
+            throw new InventoryUnavailableException("Sku is not Active");
```

Comment:
Saying that a Sku is inactive means that it's not for sale whereas the InventoryUnavailableException is really used for when a Sku is active but it doesn't have inventory at this time. It's a subtle difference but if we're going to change the signature to return something that's not an IllegalArgumentException then I think we really should create a new Exception. IMO there's a reason we're throwing an IllegalArgumentException. It's because there's no other Exception in Broadleaf that makes sense therefore we should make one